crates/pop-cli/src/commands/up/network.rs

```diff
@@ -46,315 +68,553 @@ pub(crate) struct ZombienetCommand {
+		}
+
+		// Determine network config from args.
+		let network_config = match self.path.as_ref() {
+			Some(path) => path,
+			#[allow(deprecated)]
+			None => match self.file.as_deref() {
+				None => {
+					cli.outro_cancel("🚫 A network configuration file must be specified. See `pop up network --help` for usage.".to_string())?;
+					return Ok(())
+				},
+				Some(path) => path,
+			},
+		};
+
+		spawn(
+			network_config.try_into()?,
 			self.relay_chain.as_deref(),
 			self.relay_chain_runtime.as_deref(),
 			self.system_parachain.as_deref(),
 			self.system_parachain_runtime.as_deref(),
 			self.parachain.as_ref(),
+			self.verbose,
+			self.skip_confirm,
+			self.command.as_deref(),
+			cli,
 		)
 		.await
-		{
-			Ok(n) => n,
-			Err(e) =>
-				return match e {
-					Error::Config(message) => {
-						outro_cancel(format!("🚫 A configuration error occurred: `{message}`"))?;
-						Ok(())
-					},
-					Error::MissingBinary(name) => {
-						outro_cancel(format!("🚫 The `{name}` binary is specified in the network configuration file, but cannot be resolved to a source. Are you missing a `--parachain` argument?"))?;
-						Ok(())
-					},
-					_ => Err(e.into()),
-				},
-		};
+	}
+}
 
-		// Source any missing/stale binaries
-		if Self::source_binaries(&mut zombienet, &cache, self.verbose, self.skip_confirm).await? {
-			return Ok(());
+/// Launch a local network with supported parachains.
+#[derive(Args, Clone)]
+#[cfg_attr(test, derive(Default))]
+pub(crate) struct BuildCommand<const FILTER: u8> {
+	/// The version of the binary to be used for the relay chain, as per the release tag (e.g.
+	/// "stable2503"). See <https://github.com/paritytech/polkadot-sdk/releases> for more details.
+	#[arg(short, long)]
+	relay_chain: Option<String>,
+	/// The version of the runtime to be used for the relay chain, as per the release tag (e.g.
+	/// "v1.4.1"). See <https://github.com/polkadot-fellows/runtimes/releases> for more details.
+	#[arg(short = 'R', long)]
+	relay_chain_runtime: Option<String>,
+	/// The version of the binary to be used for system parachains, as per the release tag (e.g.
+	/// "stable2503"). Defaults to the relay chain version if not specified.
+	/// See <https://github.com/paritytech/polkadot-sdk/releases> for more details.
+	#[arg(short, long)]
+	system_parachain: Option<String>,
+	/// The version of the runtime to be used for system parachains, as per the release tag (e.g.
+	/// "v1.4.1"). See <https://github.com/polkadot-fellows/runtimes/releases> for more details.
+	#[arg(short = 'S', long)]
+	system_parachain_runtime: Option<String>,
+	/// The parachain(s) to be included. An optional parachain identifier and/or port can be
+	/// affixed via #id and :port specifiers (e.g. `asset-hub#1000:9944`).
+	#[arg(short, long, value_delimiter = ',', value_parser = SupportedParachains::<FILTER>::new())]
+	parachain: Option<Vec<Box<dyn Para>>>,
+	/// The port to be used for the first relay chain validator.
+	#[clap(short = 'P', long)]
```

Comment:
This is maybe a little clumsy - as the relay is determined via a command (paseo, polkadot), there needs to be a way to specify the port for a validator. `pop up paseo:9944` doesnt seem great.